cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -186,6 +187,42 @@ struct MultiplyChecked {
   }
 };
 
+struct Divide {
+  template <typename T, typename Arg0, typename Arg1>
+  static constexpr enable_if_floating_point<T> Call(KernelContext*, Arg0 left,
+                                                    Arg1 right) {
+    return left / right;
```

Comment:
```suggestion
    if (ARROW_PREDICT_FALSE(right == 0)) {
      ctx->SetStatus(Status::Invalid("divide by zero"));
      return 0;
    }
    return left / right;
```

Reply:
Thanks for your suggestion.
According to the IEEE 754 standard, dividing by zero should be well-defined for floating point numbers. Please see, for example, https://dl.acm.org/doi/10.1145/103162.103163, page 26.

Reply:
This is true, but division by zero can result in `SIGFPE` if [`feenableexcept()`](http://www.gnu.org/savannah-checkouts/gnu/libc/manual/html_node/Control-Functions.html) or a hardware exception if [`_control87()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/control87-controlfp-control87-2) are used. For example:

```sh
g++ -lm -xc++ - <<!
#include <fenv.h>
int main(int argc, char* argv[]) {
  feenableexcept(FE_ALL_EXCEPT);
  return static_cast<int>(1.0 / 0.0);
}
!
./a.out # abort
```

These are rare, so maybe we don't need to support them (or at least restrict this check to the checked division kernel), but they do represent a potential crash when dividing by `0.0`. @pitrou ?

Reply:
I'd say if the user really wants to enable FP exceptions process-wise, we shouldn't try to counteract them.

Reply:
Alright, I withdraw this suggestion